src/tests/test_process.py

```diff
@@ -0,0 +1,112 @@
+"""
+Common reusable code for tests and tests for pybitmessage process.
+"""
+
+import os
+import signal
+import subprocess  # nosec
+import tempfile
+import unittest
+from time import sleep
+from datetime import datetime
+
+import psutil
+
+
+def put_signal_file(path, filename):
+    """Creates file, presence of which is a signal about some event."""
+    with open(os.path.join(path, filename), 'wb') as outfile:
+        outfile.write(datetime.now().isoformat())
+
+
+class TestProcessProto(unittest.TestCase):
+    """Test case implementing common logic for external testing:
+    it starts pybitmessage in setUpClass() and stops it in tearDownClass()
+    """
+    _process_cmd = ['pybitmessage', '-d']
+    _threads_count = 14
+    _files = (
+        'keys.dat', 'debug.log', 'messages.dat', 'knownnodes.dat',
+        '.api_started', 'unittest.lock'
+    )
+
+    @classmethod
+    def setUpClass(cls):
+        """Setup environment and start pybitmessage"""
+        cls.home = os.environ['BITMESSAGE_HOME'] = tempfile.gettempdir()
+        put_signal_file(cls.home, 'unittest.lock')
+        subprocess.call(cls._process_cmd)  # nosec
+        sleep(5)
+        cls.pid = int(cls._get_readline('singleton.lock'))
+        cls.process = psutil.Process(cls.pid)
+
+    @classmethod
+    def _get_readline(cls, pfile):
+        pfile = os.path.join(cls.home, pfile)
+        try:
+            return open(pfile, 'rb').readline().strip()
+        except (OSError, IOError):
+            pass
+
+    @classmethod
+    def _cleanup_files(cls):
+        for pfile in cls._files:
+            try:
+                os.remove(os.path.join(cls.home, pfile))
+            except OSError:
+                pass
+
+    @classmethod
+    def tearDownClass(cls):
+        """Ensures that pybitmessage stopped and removes files"""
+        # os.kill(cls.pid, signal.SIGTERM)
+        cls.process.send_signal(signal.SIGTERM)
+        try:
+            cls.process.wait(5)
+        except psutil.TimeoutExpired:
+            print(open(os.path.join(cls.home, 'debug.log'), 'rb').read())
+            cls.process.kill()
+        finally:
+            cls._cleanup_files()
+
+    def _test_threads(self):
```

Comment:
where?

Reply:
pylint doesn't like multiline comments as docsrings even if pep8 does. We can either be consistent (something something hobgoblin minds) or we can add a pylint exception to avoid `Missing function docstring (missing-docstring)`. I prefer the former.

Reply:
Why? Just to satisfy a lint? Don't you understand what this method does? `TestProcess.test_threads()` has docstring, it's enough as for me.

Reply:
Never mind, my bad, with all the multi-line `#` comments I missed that this method starts with `_` so the linter doesn't even care.

I was only thinking about helping you get a green codacy commit. If creating a docstring on public methods is clearly not required then we should leave a lint marker disabling the check in the most specific place possible. 'satisfying the linter' is about being able to use the return code as part of our CI and as far as humans go, 'alert-blindness' is a thing. If you're used to ignoring pages of lint warning you're making it harder to see ones that matter (more).